onboarding/views.py

```diff
@@ -760,15 +771,34 @@ class UserProgressView(viewsets.ModelViewSet):
 
     def get_queryset(self):
         user = self.request.user
-        employee = self.kwargs['user_id']
+        employee = self.kwargs.get('user_id', None)
 
         if self.request.user.is_hr:
-            # queryset = Page.objects.filter(section__answer__owner=employee, section__owner=user.company)
-            queryset = Answer.objects.select_related('section', 'section__page').filter(owner=employee, section__owner=user.company)
+            if employee is not None:
+                # queryset = Page.objects.filter(section__answer__owner=employee, section__owner=user.company)
+                queryset = Answer.objects.select_related('section', 'section__page').filter(owner=employee, section__owner=user.company)
+            else:
+                # users = User.objects.filter(company=self.request.user.company)
+                # queryset = Answer.objects.select_related('section', 'section__page').filter(owner__in=users, section__owner=user.company)
+                queryset = Answer.objects.none()
+            #
         else:
             queryset = Answer.objects.select_related('section', 'section__page').filter(owner=user, section__owner=user.company)
 
         return queryset
+
+    @action(detail=False, permission_classes=[IsAuthenticated, IsHrUser], serializer_class=UserProgressLimitedSerializer, url_name="progress-of-all")
+    def list_all(self, request, pk=None):
```

Comment:
Can you tell me why you're declaring pk=None (pk is optional argument I think)?
pk = <certain number> I would understand but None - does it change anything?

Reply:
Its common value to be assigned to,
just like `null` in Java, `NULL` in C (if it is pointer - long story).
In JavaScript you have `null` or `unsigned`.
`1` probably would be wrong because you force `url` of API to have `1` at the end by default;
`None` is like: there is something or nothing.
Sometimes I see in the codes `pk=None` or `pk` alone, but I can't make a lecture about that bec. honestly I don't fully understand where this custom came from or why exactly.
Here we want to be sure that empty `url` makes `pk` being `None`;
it is not used yet though.